File: Python algo/Arrays/10-validSudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        cubes = [
            [set(), set(), set()],
            [set(), set(), set()],
            [set(), set(), set()]
        ]
        
        for i in range(9):
            row = set()
            column = set()

            for j in range(9):
                cube = cubes[i//3][j//3]

                if board[i][j] in cube and board[i][j] != '.':
                    return False
                cube.add(board[i][j])

                if board[i][j] in row and board[i][j] != '.':
                    return False
                row.add(board[i][j])

                if board[j][i] in column and board[j][i] != '.':
                    return False
                column.add(board[i][j])

        return True

sol = Solution()

board = [["5","3",".",".","7",".",".",".","."]
        ,["6",".",".","1","9","5",".",".","."]
        ,[".","9","8",".",".",".",".","6","."]
        ,["8",".",".",".","6",".",".",".","3"]
        ,["4",".",".","8",".","3",".",".","1"]
        ,["7",".",".",".","2",".",".",".","6"]
        ,[".","6",".",".",".",".","2","8","."]
        ,[".",".",".","4","1","9",".",".","5"]
        ,[".",".",".",".","8",".",".","7","9"]]

res = sol.isValidSudoku(board)

File: js algo/easy/arrays/10-validSudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
 var isValidSudoku = function(board) {

    var cubes = [
        [new Set(), new Set(), new Set()],
        [new Set(), new Set(), new Set()],
        [new Set(), new Set(), new Set()]
    ]

    for (let i = 0; i < 9; i++) {

        var row = new Set()
        var column = new Set()
        

        for (let j = 0; j < 9; j++) {

            var cube = cubes[parseInt(i/3)][parseInt(j/3)]

            if(cube.has(board[i][j]) && board[i][j] !== '.')
                return false
            cube.add(board[i][j])

            if(row.has(board[i][j]) && board[i][j] !== '.')
                return false
            row.add(board[i][j])

            if(column.has(board[j][i]) && board[j][i] !== '.')
                return false
                column.add(board[i][j])
        }
        
    }

    return true
};